### ISSUE TITLE
Expose Buf instead of ChannelBuffer in storehaus-redis

### ISSUE THREAD

**Issue by BenFradet** [twitter/storehaus#310]
Since Finagle is moving away from exposing netty's `ChannelBuffer` in their API in favor of twitter util's `Buf`, I think we should too.

From the recent commits, I think the replacement should be complete by Finagle's next release.

**Comment by johnynek** [comment]
Yeah, I think this is a good call. (PS: lesson to be learned here, it is a good idea to hide all your dependencies on your public API in general. We didn't do that in a few places in scalding, I really regret it).
